tests/samplers_tests/test_samplers.py

```diff
@@ -1016,8 +1018,29 @@ def objective(trial: Trial) -> float:
     sequence_dict[k] = list(study.trials[-1].params.values())
 
 
+@pytest.fixture
+def unset_seed_in_test(request: SubRequest) -> None:
+    # Unset the hashseed at beginning and restore it at end regardless of an exception in the test.
+    # See https://docs.pytest.org/en/stable/how-to/fixtures.html#adding-finalizers-directly
+    # for details.
+
+    hash_seed = os.getenv("PYTHONHASHSEED")
+    if hash_seed is not None:
+        del os.environ["PYTHONHASHSEED"]
+
+    def restore_seed() -> None:
+        if hash_seed is not None:
+            os.environ["PYTHONHASHSEED"] = hash_seed
+
+    request.addfinalizer(restore_seed)
+
+
 @pytest.mark.parametrize("sampler_class_index", range(len(sampler_class_with_seed)))
-def test_reproducible_in_other_process(sampler_class_index: int) -> None:
+def test_reproducible_in_other_process(sampler_class_index: int, unset_seed_in_test: None) -> None:
```

Comment:
I have two questions about type hint. `unset_seed_in_test` is not referred to in this test function, and I guess why `mypy` raise any error without any annotations. I know that it is a fixture and is called background.

And also, the type of `unset_seed_in_test` is set to `None`. Is it a return type of `unset_seed_in_test()`?

Reply:
Yes, your understanding is correct.

Reply:
Thank you.